### PR TITLE
feat: scoped var support in cm/cs

### PR DESCRIPTION
**Pull request by ritvikmahajan17**
# Description

Scoped variables support in CM:
1. Resolved scoped variables in editor
Screenshot 2023-10-30 at 1 03 55 PM

2. Floating widget
Screenshot 2023-10-30 at 1 04 11 PM

Scoped variables support in CS:
1. Resolved scoped variables in editor
Screenshot 2023-10-30 at 1 04 01 PM

2. Floating widget
Screenshot 2023-10-30 at 1 04 15 PM

Fixes #4095

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/app/details/cdDetails/service.ts [hunk 1]
```diff
@@ -137,12 +137,20 @@ export const prepareConfigMapAndSecretData = (
         } else {
             secretValues['external'] = { displayName: 'Data type', value: EXTERNAL_TYPES[type][''] }
             if (type === 'Secret' && historyData.codeEditorValue.value) {
-                const secretData = JSON.parse(historyData.codeEditorValue.value)
-                const decodeNotRequired =
-                    skipDecode || Object.keys(secretData).some((data) => secretData[data] === '*****') // Don't decode in case of non admin user
+                const secretData_value = JSON.parse(historyData.codeEditorValue.value)
```
Comment (vivek-devtron) [inline src/components/app/details/cdDetails/service.ts:140]: Please use camelCase variable naming convention

Comment (ritvikmahajan17) [reply src/components/app/details/cdDetails/service.ts]: resolved

File: src/components/app/details/cdDetails/service.ts [hunk 2]
```diff
@@ -138,11 +138,18 @@ export const prepareConfigMapAndSecretData = (
...
                 const secretData = JSON.parse(historyData.codeEditorValue.value)
-                const decodeNotRequired =
+                const resolvedSecretData = JSON.parse(historyData.codeEditorValue.resolvedValue)
+                let decodeNotRequired =
                     skipDecode || Object.keys(secretData).some((data) => secretData[data] === '*****') // Don't decode in case of non admin user
                 historyData.codeEditorValue.value = decodeNotRequired
                     ? historyData.codeEditorValue.value
                     : JSON.stringify(decode(secretData))
+
+                decodeNotRequired =
```
Comment (vivek-devtron) [inline src/components/app/details/cdDetails/service.ts:148]: We can use the above value. No need to recalculate the same

Comment (ritvikmahajan17) [reply src/components/app/details/cdDetails/service.ts]: resolved

**Review by vivek-devtron — DISMISSED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**